From 9669f10ef1cc954a6d37a054ac0b2b434099053d Mon Sep 17 00:00:00 2001
From: Michael Campion <michael.campion@emutex.com>
Date: Wed, 20 Nov 2019 11:33:52 +0000
Subject: [PATCH 1/3] spi: pxa2xx: Set dir and value when requesting cs gpio

The pxa2xx driver assumes the gpio direction and value have already been
configured prior to requesting the gpio for chip select.

This is not always the case on the UP Platform so this commit replaces
GPIOD_ASIS with GPIOD_OUT_HIGH.

Signed-off-by: Michael Campion <michael.campion@emutex.com>
---
 drivers/spi/spi-pxa2xx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/spi/spi-pxa2xx.c b/drivers/spi/spi-pxa2xx.c
index 4cb515a..1f54a7b 100644
--- a/drivers/spi/spi-pxa2xx.c
+++ b/drivers/spi/spi-pxa2xx.c
@@ -1769,7 +1769,7 @@ static int pxa2xx_spi_probe(struct platform_device *pdev)
 		for (i = 0; i < master->num_chipselect; i++) {
 			struct gpio_desc *gpiod;
 
-			gpiod = devm_gpiod_get_index(dev, "cs", i, GPIOD_ASIS);
+			gpiod = devm_gpiod_get_index(dev, "cs", i, GPIOD_OUT_HIGH);
 			if (IS_ERR(gpiod)) {
 				/* Means use native chip select */
 				if (PTR_ERR(gpiod) == -ENOENT)
-- 
2.7.4

